# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1485468399 18000
#      Thu Jan 26 17:06:39 2017 -0500
# Node ID 5a8979c8d0c3e96e0d29effb4a61514e3ac27459
# Parent  28c9a42cfbf85d85ec6139864a3e9ba68f1591f1
call octave_quit after return from "foreign" code

* quit.h (END_INTERRUPT_IMMEDIATELY_IN_FOREIGN_CODE): Call octave_quit.

diff --git a/liboctave/cruft/misc/quit.h b/liboctave/cruft/misc/quit.h
--- a/liboctave/cruft/misc/quit.h
+++ b/liboctave/cruft/misc/quit.h
@@ -250,26 +250,27 @@ inline void octave_quit (void)
                                                                 \
       octave_save_current_context (saved_context);              \
                                                                 \
       if (octave_set_current_context)                           \
         {                                                       \
           octave_restore_current_context (saved_context)
 
 #define BEGIN_INTERRUPT_IMMEDIATELY_IN_FOREIGN_CODE_2   \
-  }                                                     \
+        }                                               \
       else                                              \
         {                                               \
-  octave_interrupt_immediately++
+          octave_interrupt_immediately++
 
-#define END_INTERRUPT_IMMEDIATELY_IN_FOREIGN_CODE       \
-  octave_interrupt_immediately--;                       \
-  octave_restore_current_context (saved_context);       \
-}                                                       \
-}                                                       \
+#define END_INTERRUPT_IMMEDIATELY_IN_FOREIGN_CODE               \
+          octave_interrupt_immediately--;                       \
+          octave_restore_current_context (saved_context);       \
+          octave_quit ();                                       \
+        }                                                       \
+    }                                                           \
   while (0)
 
 #if defined (__cplusplus)
 
 #define BEGIN_INTERRUPT_WITH_EXCEPTIONS                                 \
   sig_atomic_t saved_octave_interrupt_immediately = octave_interrupt_immediately; \
                                                                         \
   try                                                                   \
